### DISCUSSION TITLE
Issue in rendering tables and sub-figures in word

### DISCUSSION THREAD

**Discussion by Micbut** [quarto-dev/quarto-cli#9899] [Q&A]
### Description

Hi,
I am using Quarto 1.5.42 on a Windows 10 Enterprise 22H2

I am having issues in rendering tables and sub-figures

**The issue with sub-figures is the following**
I use this code:
```
::: {#fig-room layout-ncol="2"}
![set up with lights](images/room_1.png){#fig-a height="300"}
![rendered set up](images/room_2.png){#fig-b height="300"}
The empty room
:::
```
The following is the rendering in HTML and it looks good

![image](https://github.com/quarto-dev/quarto-cli/assets/54619028/9d2c72ec-7557-4c6f-a7c7-59a7f6c8fd9f)

This is the outcome in word

![image](https://github.com/quarto-dev/quarto-cli/assets/54619028/bc072ea3-4950-47d8-ab14-c4ca32eae135)

**The issue with the table is the following:**

I have this table
```
| Process                 | Description | Meristem | Internode | Cotyledon | Leaf | Truss |
|:----------|-----------|-----------|-----------|-----------|-----------|-----------|
| Maintenance respiration |             | \-       | \+        | \+        | \+   | \+    |
| Growth Respiration      |             | \-       | \+        | \+        | \+   | \+    |
| Photosynthesis          |             | \-       | \-        | \+        | \+   | \-    |
| Potential Growth        |             | \-       | \+        | \-        | \+   | \+    |
| Actual Growth           |             | \-       | \+        | \+        | \+   | \+    |
```

: Overview of the processes for different organ types when running the dynamic FSP model. {#tbl-overview_processes}

This is how the table look like in word
![image](https://github.com/quarto-dev/quarto-cli/assets/54619028/4e0159af-c84a-484b-a921-bcd677e8a1dd)

However, in scietific pubblications, table should have a horizontal delimitation bar also at the bottom, and I don't manage to generate it

Another issue I noticed, is also that the text doesn't align on the left of the table when rendered in word

I would really be happy if someone can help me out, thank you so much!

**Comment by mcanouil** [comment]
Could you share a complete small reproducible example?
Since you are using 1.5, you could use `{{< placeholder >}}` to get images or the URL as below.

This being said I believe your issue is the one reported in:
- https://github.com/quarto-dev/quarto-cli/issues/9623

Table styling in Word is defined by the template: https://quarto.org/docs/output-formats/ms-word-templates.html

Don't hesitate to look in issues/discussions for example using keywords or even the labels: https://github.com/quarto-dev/quarto-cli/labels/docx https://github.com/quarto-dev/quarto-cli/labels/layout etc.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by Micbut** [comment]
**Quarto Check**

````
$ quarto check
Quarto 1.5.42
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.42
      Path: C:\Users\buttu001\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: v2024.03.13
      Chromium: 869685

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\buttu001\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.8.2
      Path: C:/Program Files/Python38/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

(\) Checking R installation...........Warning message:
In normalizePath(path.expand(path), winslash, mustWork) :
  path[1]="//WURNET.NL/Homes/buttu001/My Documents": Access is denied
(|) Checking R installation...........Warning message:
In normalizePath(path.expand(path), winslash, mustWork) :
  path[1]="//WURNET.NL/Homes/buttu001/My Documents": Access is denied
[>] Checking R installation...........OK
      Version: 4.2.0
      Path: C:/Users/buttu001/AppData/Local/Programs/R/R-4.2.0
      LibPaths:
        - C:/Users/buttu001/AppData/Local/Programs/R/R-4.2.0/library
      knitr: 1.41
      rmarkdown: 2.25

[>] Checking Knitr engine render......OK
````
**small reproducible example**
````

---
title: "Chapter 2 - FSPM Models Description"
toc: true
number-sections: true
format:
  html:
    output-file: FSPM Models Description.html
  docx:
    output-file: FSPM Models Description.docx
---


| Process                 | Description | Meristem | Internode | Cotyledon | Leaf | Truss |
|:----------|-----------|-----------|-----------|-----------|-----------|-----------|
| Maintenance respiration |             | \-       | \+        | \+        | \+   | \+    |
| Growth Respiration      |             | \-       | \+        | \+        | \+   | \+    |
| Photosynthesis          |             | \-       | \-        | \+        | \+   | \-    |
| Potential Growth        |             | \-       | \+        | \-        | \+   | \+    |
| Actual Growth           |             | \-       | \+        | \+        | \+   | \+    |

: Overview of the processes for different organ types when running the dynamic FSP model. {#tbl-overview_processes}


::: {#fig-room layout-ncol="2"}
![An image](https://placehold.co/600x400.png)

![An image](https://placehold.co/600x400.png)

The empty room
:::
````

**Comment by Micbut** [comment]
Furthermore, about using the custom word template;
I made this document, and created this table style - so that when clicking on create table when using word the table is create as I want. The table style is called Style1

[custom-reference-doc.docx](https://github.com/user-attachments/files/15637748/custom-reference-doc.docx)

However, I do not manage to apply it. This is what I tried, and didn't work

````
---
title: "Chapter 2 - FSPM Models Description"
toc: true
number-sections: true
format:
  html:
    output-file: FSPM Models Description.html
  docx:
    reference-doc: custom-reference-doc.docx
    output-file: FSPM Models Description.docx

execute:
  echo: false
  warning: false
  message: false
---
::: {custom-style="Style1"}
| Process                 | Description | Meristem | Internode | Cotyledon | Leaf | Truss |
|:----------|-----------|-----------|-----------|-----------|-----------|-----------|
| Maintenance respiration |             | \-       | \+        | \+        | \+   | \+    |
| Growth Respiration      |             | \-       | \+        | \+        | \+   | \+    |
| Photosynthesis          |             | \-       | \-        | \+        | \+   | \-    |
| Potential Growth        |             | \-       | \+        | \-        | \+   | \+    |
| Actual Growth           |             | \-       | \+        | \+        | \+   | \+    |

: Overview of the processes for different organ types when running the dynamic FSP model. {#tbl-overview_processes}
:::
````

**Comment by Micbut** [comment]
Furthermore, even if I managed to customize the headings using the template document, the following commands don t work in my case. The command was taken from the example from here: https://quarto.org/docs/output-formats/ms-word-templates.html
````
---
title: "Chapter 2 - FSPM Models Description"
toc: true
number-sections: true
format:
  html:
    output-file: FSPM Models Description.html
  docx:
    reference-doc: custom-reference-doc.docx
    output-file: FSPM Models Description.docx

execute:
  echo: false
  warning: false
  message: false
---

[Get out]{custom-style="Emphasis"}, he said.

::
: {custom-style="Poetry"}
| A Bird came down the Walk---
| He did not know I saw---
:::
````
the outcome is not customized

![Screenshot 2024-06-06 162141](https://github.com/quarto-dev/quarto-cli/assets/54619028/9a711dab-240b-407e-a085-ad9e32ee65e2)

**Reply by mcanouil** [reply]
That's not how it works.

If you want to do things like `[Get out]{custom-style="Emphasis"}` then you need to write a Lua filter and use OpenXML code.

For example, see the "highlight text" Quarto extension (https://github.com/mcanouil/quarto-highlight-text), especially the following part: https://github.com/mcanouil/quarto-highlight-text/blob/main/_extensions/highlight-text/highlight-text.lua#L72-L86

You cannot define custom style in the template. Pandoc does not know about them.
You can only modify what already exist and following our (or Pandoc) guide.

**Reply by cderv** [reply]
> You cannot define custom style in the template. Pandoc does not know about them.

I don't think that is true. Pandoc allows to define and use custom styles - See https://pandoc.org/MANUAL.html#custom-styles

Is this not working in quarto ? it should.

**Reply by mcanouil** [reply]
Thanks @cderv so if this is not working one possible reason is that the style is not available in the reference styles within the template (or possibly a bug).

**Comment by Micbut** [comment]
Thank you for your quick answer, I misunderstood how to use the word template. However, I still don't manage to get the tables as I needed using the [custom-reference-doc.docx](https://github.com/user-attachments/files/15637744/custom-reference-doc.docx)

Table with Style1 is the one I have set as default, but when I render in Quarto the table is not correctly generated
